<!DOCTYPE html>
<html>
<head>
  <title>CRAN Task View: Infectious Disease Epidemiology</title>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8" />
  <meta name="citation_title" content="CRAN Task View: Infectious Disease Epidemiology" />
  <meta name="citation_author" content="Thibaut Jombart, Matthieu Rolland, Hugo Gruson" />
  <meta name="citation_publication_date" content="2021-02-25" />
  <meta name="DC.title" content="CRAN Task View: Infectious Disease Epidemiology" />
  <meta name="DC.creator" content="Thibaut Jombart, Matthieu Rolland, Hugo Gruson" />
  <meta name="DC.issued" content="2021-02-25" />
</head>

<body>
  <h2>CRAN Task View: Infectious Disease Epidemiology</h2>
  <table>
    <tr><td valign="top"><b>Maintainer:</b></td><td>Thibaut Jombart, Matthieu Rolland, Hugo Gruson</td></tr>
    <tr><td valign="top"><b>Contact:</b></td><td>&#x74;&#x68;&#x69;&#x62;&#x61;&#x75;&#x74;&#x6a;&#x6f;&#x6d;&#x62;&#x61;&#x72;&#x74;&#x20;&#x61;&#x74;&#x20;&#x67;&#x6d;&#x61;&#x69;&#x6c;&#x2e;&#x63;&#x6f;&#x6d;</td></tr>
    <tr><td valign="top"><b>Version:</b></td><td>2021-02-25</td></tr>
    <tr><td valign="top"><b>Citation:</b></td><td>Thibaut Jombart, Matthieu Rolland, Hugo Gruson (2021). CRAN Task View: Infectious Disease Epidemiology. Version 2021-02-25.</td></tr>
  </table>

<p>Contributors (in alphabetic order): Neale Batra, Rich FitzJohn, Hugo Gruson, Andreas Handel, Michael Höhle, Thibaut Jombart, Joseph Larmarange, Sebastian Lequime, Alex Spina, Tim Taylor, Sean Wu</p><h2 id="overview">Overview</h2><p>R is increasingly becoming a standard in infectious disease epidemiology (IDE), providing a wide array of tools covering different aspects of outbreak analytics from study design to data exploration, modelling, forecasting and simulation. This task view provides an overview of packages specifically developed for IDE. It does not encompass other applications of epidemiology (such as environmental epidemiology) or cross-cutting tools used in, but not specific to, IDE.</p><p>Packages were regrouped in the following categories:</p><ol type="1"><li><strong>Data exploration and visualisation:</strong> these tools are dedicated to handling and visualising IDE data, producing epidemic curves (‘<em>epicurves</em>’), exploring contact tracing data, etc.</li><li><strong>Infectious disease modelling:</strong> these tools encompass various approaches for the analysis of epidemic curves (including outbreak detection / surveillance), estimating reproduction numbers (<em>R</em>), providing short-term forecasting, implementing compartmental models (e.g. SIR models), simulating outbreaks and reconstructing transmission trees</li><li><strong>Helpers</strong>: tools implementing miscellaneous tasks useful in IDE</li><li><strong>Data packages</strong>: these packages provide access to both empirical and simulated epidemic data; includes a specific section on COVID-19.</li></ol><p>Additional links to non specific but highly useful packages (to create tables, manipulate dates, etc.) are provided in the task view’s footnotes.</p><h2 id="inclusion-criteria">Inclusion criteria</h2><p>Packages included in this task view were identified through recommendations of IDE experts as well as an automated CRAN search using <code>pkgsearch::pkg_search()</code>, with the keywords: <em>epidemiology</em>, <em>epidemic</em>, <em>epi</em>, <em>outbreak</em> and <em>transmission</em>. The list was manually curated for the final selection to satisfy the conditions described in the previous paragraph.</p><p>Packages are deemed in scope if they provide tools, or data, explicitly targeted at reporting, modelling, or forecasting infectious diseases.</p><p><strong>Your input is welcome!</strong> Please suggest packages we may have missed by posting an issue at:</p><p>https://github.com/bisaloo/InfectiousEpi</p><h2 id="data-visualisation-and-exploration">Data visualisation and exploration</h2><p>Here are packages providing help for visualising and exploring epidemic data (plotting epidemic curves, incidence curves, contact tracing, etc.)</p><ul><li><a href="https://CRAN.R-project.org/package=epicontacts">epicontacts</a>: A collection of tools for representing epidemiological contact data, composed of case line lists and contacts between cases. Also contains procedures for data handling, interactive graphics, and statistics.</li><li><a href="https://CRAN.R-project.org/package=EpiContactTrace">EpiContactTrace</a>: Routines for epidemiological contact tracing and visualisation of network of contacts.</li><li><a href="https://CRAN.R-project.org/package=EpiCurve">EpiCurve</a>: Creates simple or stacked epidemic curves for hourly, daily, weekly or monthly outcome data.</li><li><a href="https://CRAN.R-project.org/package=epiDisplay">epiDisplay</a>: Package for data exploration and result presentation.</li><li><a href="https://CRAN.R-project.org/package=epiflows">epiflows</a>: Provides functions and classes designed to handle and visualise epidemiological flows of people between locations. Also contains a statistical method for predicting disease spread from flow data initially described in <a href="https://doi.org/10.2807%2F1560-7917.ES.2017.22.28.30572">Dorigatti et al. (2017)</a>. This package is part of the <a href="https://www.repidemicsconsortium.org/">RECON</a> toolkit for outbreak analysis.</li><li><a href="https://CRAN.R-project.org/package=EpiReport">EpiReport</a>: Drafting an epidemiological report in ‘Microsoft Word’ format for a given disease, similar to the Annual Epidemiological Reports published by the European Centre for Disease Prevention and Control.</li><li><a href="https://CRAN.R-project.org/package=i2extras">i2extras</a>: Provides functions to work with ‘incidence2’ objects, including a simplified interface for trend fitting, estimation of growth rates, and peak estimation. This package is part of the <a href="https://www.repidemicsconsortium.org/">RECON</a> toolkit for outbreak analysis.</li><li><a href="https://CRAN.R-project.org/package=incidence">incidence</a>: Provides functions and classes to compute, handle and visualise incidence from dated events for a defined time interval. Dates can be provided in various standard formats. The class ‘incidence’ is used to store computed incidence and can be easily manipulated, subsetted, and plotted. In addition, log-linear models can be fitted to ‘incidence’ objects using ‘fit’. This package is part of the <a href="https://www.repidemicsconsortium.org/">RECON</a> toolkit for outbreak analysis, but is now scheduled for deprecation and is replaced by <a href="https://CRAN.R-project.org/package=incidence2">incidence2</a>.</li><li><a href="https://CRAN.R-project.org/package=incidence2">incidence2</a>: Provides functions and classes to compute, handle and visualise incidence from dated events for a defined time interval. Improves the original <a href="https://CRAN.R-project.org/package=incidence">incidence</a> package in many ways: full flexibility in time intervals used, allows multiple stratification, and is fully compatible with <a href="https://CRAN.R-project.org/package=dplyr">dplyr</a> and other tidyverse tools. This package is part of the <a href="https://www.repidemicsconsortium.org/">RECON</a> toolkit for outbreak analysis.</li></ul><h2 id="infectious-disease-modelling">Infectious disease modelling</h2><h3 id="epidemics-surveillance">Epidemics surveillance</h3><ul><li><a href="https://CRAN.R-project.org/package=argo">argo</a>: Augmented Regression with General Online data (ARGO) for accurate estimation of influenza epidemics in United States on both national level and regional level. It replicates the method introduced in paper <a href="https://doi.org/10.1073/pnas.1515373112">Yang, S., Santillana, M. and Kou, S.C. (2015)</a> and <a href="https://doi.org/10.1038/s41598-019-41559-6">Ning, S., Yang, S. and Kou, S.C. (2019)</a>.</li><li><a href="https://CRAN.R-project.org/package=Epi">Epi</a>: Functions for demographic and epidemiological analysis in the Lexis diagram, i.e. register and cohort follow-up data, in particular representation, manipulation and simulation of multistate data - the Lexis suite of functions, which includes interfaces to <a href="https://CRAN.R-project.org/package=mstate">mstate</a>, <a href="https://CRAN.R-project.org/package=etm">etm</a> and <a href="https://CRAN.R-project.org/package=cmprsk">cmprsk</a> packages. Also contains functions for Age-Period-Cohort and Lee-Carter modeling and a function for interval censored data and some useful functions for tabulation and plotting, as well as a number of epidemiological data sets.</li><li><a href="https://CRAN.R-project.org/package=episensr">episensr</a>: Basic sensitivity analysis of the observed relative risks adjusting for unmeasured confounding and misclassification of the exposure/outcome, or both. It follows the bias analysis methods and examples from the book by Lash T.L, Fox M.P, and Fink A.K. “Applying Quantitative Bias Analysis to Epidemiologic Data”, (‘Springer’, 2009).</li><li><a href="https://CRAN.R-project.org/package=mem">mem</a>: The Moving Epidemic Method, created by T Vega and JE Lozano (<a href="https://doi.org/10.1111/j.1750-2659.2012.00422.x">2012</a>, <a href="https://doi.org/10.1111/irv.12330">2015</a>), allows the weekly assessment of the epidemic and intensity status to help in routine respiratory infections surveillance in health systems. Allows the comparison of different epidemic indicators, timing and shape with past epidemics and across different regions or countries with different surveillance systems. Also, it gives a measure of the performance of the method in terms of sensitivity and specificity of the alert week.</li><li><a href="https://CRAN.R-project.org/package=memapp">memapp</a>: The Moving Epidemic Method, created by T Vega and JE Lozano (<a href="https://doi.org/10.1111/j.1750-2659.2012.00422.x">2012</a>, <a href="https://doi.org/10.1111/irv.12330">2015</a>), allows the weekly assessment of the epidemic and intensity status to help in routine respiratory infections surveillance in health systems. Allows the comparison of different epidemic indicators, timing and shape with past epidemics and across different regions or countries with different surveillance systems. Also, it gives a measure of the performance of the method in terms of sensitivity and specificity of the alert week. ‘memapp’ is a web application created in the Shiny framework for the <a href="https://CRAN.R-project.org/package=mem">mem</a> R package.</li><li><a href="https://CRAN.R-project.org/package=nosoi">nosoi</a>: The aim of <a href="https://CRAN.R-project.org/package=nosoi">nosoi</a> (pronounced no.si) is to provide a flexible agent-based stochastic transmission chain/epidemic simulator (<a href="https://doi.org/2020.03.03.973107">Lequime et al. 2020</a>). It is named after the daimones of plague, sickness and disease that escaped Pandora’s jar in the Greek mythology. <a href="https://CRAN.R-project.org/package=nosoi">nosoi</a> is able to take into account the influence of multiple variable on the transmission process (e.g. dual-host systems (such as arboviruses), within-host viral dynamics, transportation, population structure), alone or taken together, to create complex but relatively intuitive epidemiological simulations.</li><li><a href="https://CRAN.R-project.org/package=riskCommunicator">riskCommunicator</a>: Estimates flexible epidemiological effect measures including both differences and ratios using the parametric G-formula developed as an alternative to inverse probability weighting. It is useful for estimating the impact of interventions in the presence of treatment-confounder-feedback. G-computation was originally described by <a href="https://doi.org/10.1016/0270-0255(86)90088-6">Robbins (1986)</a> and has been described in detail by <a href="https://doi.org/10.1093/aje/kwp015">Ahern, Hubbard, and Galea (2009)</a>; <a href="https://doi.org/10.1093/aje/kwq472">Snowden, Rose, and Mortimer (2011)</a>; and <a href="https://doi.org/10.1002/sim.5316">Westreich et al. (2012)</a>.</li><li><a href="https://CRAN.R-project.org/package=RSurveillance">RSurveillance</a>: Provides a diverse set of functions useful for the design and analysis of disease surveillance activities.</li><li><a href="https://CRAN.R-project.org/package=trendeval">trendeval</a>: Provides a coherent interface for evaluating models fit with the trending package. This package is part of the <a href="https://www.repidemicsconsortium.org/">RECON</a> toolkit for outbreak analysis.</li><li><a href="https://CRAN.R-project.org/package=SpatialEpi">SpatialEpi</a>: Methods and data for cluster detection and disease mapping.</li><li><a href="https://CRAN.R-project.org/package=surveillance">surveillance</a>: Statistical methods for the modeling and monitoring of time series of counts, proportions and categorical data, as well as for the flexible covariate based regression modeling of spatio-temporal point processes of epidemic phenomena. The monitoring methods focus on aberration detection in count data time series from public health surveillance of communicable diseases. A recent overview of the available monitoring procedures is given by <a href="https://doi.org/10.18637%2Fjss.v070.i10">Salmon et al. (2016)</a> and a recent overview of the implemented space-time modeling frameworks for epidemic phenomena is given by <a href="https://doi.org/10.18637%2Fjss.v077.i11">Meyer et al. (2017)</a>. Other package feature contain back-projection methods to infer time series of exposure from disease onset and correction of observed time series for reporting delays (nowcasting).</li><li><a href="https://github.com/reconhub/trendbreaker">trendbreaker</a> implements tools for detecting changes in temporal trends of a single response variable. It implements the <strong>A</strong>utomatic <strong>S</strong>election of <strong>M</strong>odels and <strong>O</strong>utlier <strong>De</strong>tection for <strong>E</strong>pidemmics (ASMODEE), an algorithm originally designed for detecting changes in COVID-19 case incidence.</li><li><a href="https://CRAN.R-project.org/package=trending">trending</a>: Provides a coherent interface to multiple modelling tools for fitting trends along with a standardised approach for generating confidence and prediction intervals.</li></ul><h3 id="estimation-of-transmissibility">Estimation of transmissibility</h3><ul><li><a href="https://CRAN.R-project.org/package=earlyR">earlyR</a>: Implements a simple, likelihood-based estimation of the reproduction number (R0) using a branching process with a Poisson likelihood. This model requires knowledge of the serial interval distribution, and dates of symptom onsets. Infectiousness is determined by weighting R0 by the probability mass function of the serial interval on the corresponding day. It is a simplified version of the model introduced by <a href="https://doi.org/10.1093%2Faje%2Fkwt133">Cori et al. (2013)</a>.</li><li><a href="https://CRAN.R-project.org/package=endtoend">endtoend</a>: Computes the expectation of the number of transmissions and receptions considering an End-to-End transport model with limited number of retransmissions per packet. It provides theoretical results and also estimated values based on Monte Carlo simulations. It is also possible to consider random data and ACK probabilities.</li><li><a href="https://CRAN.R-project.org/package=EpiEstim">EpiEstim</a>: Tools to quantify transmissibility throughout an epidemic from the analysis of time series of incidence as described in <a href="https://doi.org/10.1093/aje/kwt133">Cori et al. (2013)</a> and <a href="https://doi.org/10.1093/aje/kwh255">Wallinga and Teunis (2004)</a>.</li><li><a href="https://CRAN.R-project.org/package=epimdr">epimdr</a>: Functions, data sets and shiny apps for “Epidemics: Models and Data in R” by Ottar N. Bjornstad (<a href="https://www.springer.com/gp/book/9783319974866">ISBN 978-3-319-97487-3</a>). The package contains functions to study the S(E)IR model, spatial and age-structured SIR models; time-series SIR and chain-binomial stochastic models; catalytic disease models; coupled map lattice models of spatial transmission and network models for social spread of infection. The package is also an advanced quantitative companion to the <a href="https://www.coursera.org/learn/epidemics">coursera Epidemics Massive Online Open Course</a>.</li><li><a href="https://CRAN.R-project.org/package=epinet">epinet</a>: A collection of epidemic/network-related tools. Simulates transmission of diseases through contact networks. Performs Bayesian inference on network and epidemic parameters, given epidemic data.</li><li><a href="https://github.com/epiforecasts/EpiNow2">EpiNow2</a>: This package estimates the time-varying reproduction number, rate of spread, and doubling time using a range of open-source tools (<a href="https://doi.org/10.12688/wellcomeopenres.16006.1">Abbott et al.</a>), and current best practices (<a href="https://doi.org/10.1371/journal.pcbi.1008409">Gostic et al.</a>). It aims to help users avoid some of the limitations of naive implementations in a framework that is informed by community feedback and is under active development.</li><li><a href="https://CRAN.R-project.org/package=nbTransmission">nbTransmission</a>: Estimates the relative transmission probabilities between cases in an infectious disease outbreak or cluster using naive Bayes. Included are various functions to use these probabilities to estimate transmission parameters such as the generation/serial interval and reproductive number as well as finding the contribution of covariates to the probabilities and visualizing results. The ideal use is for an infectious disease dataset with metadata on the majority of cases but more informative data such as contact tracing or pathogen whole genome sequencing on only a subset of cases. For a detailed description of the methods see <a href="https://doi.org/10.1093%2Fije%2Fdyaa031">Leavitt et al. (2020)</a>.</li><li><a href="https://CRAN.R-project.org/package=R0">R0</a>: Estimation of reproduction numbers for disease outbreak, based on incidence data. The R0 package implements several documented methods. It is therefore possible to compare estimations according to the methods used. Depending on the methods requested by user, basic reproduction number (commonly denoted as R0) or real-time reproduction number (referred to as R(t)) is computed, along with a 95% Confidence Interval. Plotting outputs will give different graphs depending on the methods requested : basic reproductive number estimations will only show the epidemic curve (collected data) and an adjusted model, whereas real-time methods will also show the R(t) variations throughout the outbreak time period. Sensitivity analysis tools are also provided, and allow for investigating effects of varying Generation Time distribution or time window on estimates.</li><li><a href="https://CRAN.R-project.org/package=tsiR">tsiR</a>: The TSIR modeling framework allows users to fit the time series SIR model to cumulative case data, which uses a regression equation to estimate transmission parameters based on differences in cumulative cases between reporting periods. The package supports inference on TSIR parameters using GLMs and profile likelihood techniques, as well as forward simulation based on a fitted model. The package is described in <a href="https://doi.org/10.1371/journal.pone.0185528">Becker and Grenfell (2017)</a>.</li></ul><h3 id="compartmental-models">Compartmental models</h3><ul><li><a href="https://CRAN.R-project.org/package=EpiILM">EpiILM</a>: Provides tools for simulating from discrete-time individual level models for infectious disease data analysis. This epidemic model class contains spatial and contact-network based models with two disease types: Susceptible-Infectious (SI) and Susceptible-Infectious-Removed (SIR).</li><li><a href="https://CRAN.R-project.org/package=EpiILMCT">EpiILMCT</a>: Provides tools for simulating from continuous-time individual level models of disease transmission, and carrying out infectious disease data analyses with the same models. The epidemic models considered are distance-based and/or contact network-based models within Susceptible-Infectious-Removed (SIR) or Susceptible-Infectious-Notified-Removed (SINR) compartmental frameworks. An overview of the implemented continuous-time individual level models for epidemics is given by <a href="https://doi.org/10.1515/ijb-2017-0092">Almutiry and Deardon (2019)</a>.</li><li><a href="https://CRAN.R-project.org/package=EpiModel">EpiModel</a>: Tools for simulating mathematical models of infectious disease dynamics. Epidemic model classes include deterministic compartmental models, stochastic individual-contact models, and stochastic network models. Network models use the robust statistical methods of exponential-family random graph models (ERGMs) from the Statnet suite of software packages in R. Standard templates for epidemic modeling include SI, SIR, and SIS disease types. EpiModel features an API for extending these templates to address novel scientific research aims. Full methods for EpiModel are detailed in <a href="https://doi.org/10.18637/jss.v084.i08">Jenness et al. (2018)</a>.</li><li><a href="https://CRAN.R-project.org/package=odin">odin</a>: Generate systems of ordinary differential equations (ODE) and integrate them, using a domain specific language (DSL). The DSL uses R’s syntax, but compiles to C in order to efficiently solve the system. A solver is not provided, but instead interfaces to the packages <a href="https://CRAN.R-project.org/package=deSolve">deSolve</a> and <a href="https://CRAN.R-project.org/package=dde">dde</a> are generated. With these, while solving the differential equations, no allocations are done and the calculations remain entirely in compiled code. Alternatively, a model can be transpiled to R for use in contexts where a C compiler is not present. After compilation, models can be inspected to return information about parameters and outputs, or intermediate values after calculations. <a href="https://CRAN.R-project.org/package=odin">odin</a> is not targeted at any particular domain and is suitable for any system that can be expressed primarily as mathematical expressions. Additional support is provided for working with delays (delay differential equations, DDE), using interpolated functions during interpolation, and for integrating quantities that represent arrays.</li><li><a href="https://CRAN.R-project.org/package=pomp">pomp</a> Provides a large set of forward simulation algorithms and MLE or Bayesian inference techniques to work with state-space models. Models may be specified as either deterministic or stochastic and typically follow a compartmental model structure. Time may be either discrete or continuous, depending on the simulation algorithm chosen. Additionally models may be programmed in C and compiled on-the-fly into a format suitable for use in the package to speed up simulation and inference. The R package and some of its algorithms are described in <a href="https://doi.org/10.18637/jss.v069.i12">King, Nguyen, and Ionides (2016)</a>.</li><li><a href="https://CRAN.R-project.org/package=popEpi">popEpi</a>: Enables computation of epidemiological statistics, including those where counts or mortality rates of the reference population are used. Currently supported: excess hazard models, rates, mean survival times, relative survival, and standardized incidence and mortality ratios (SIRs/SMRs), all of which can be easily adjusted for by covariates such as age. Fast splitting and aggregation of ‘Lexis’ objects (from package <a href="https://CRAN.R-project.org/package=Epi">Epi</a> and other computations achieved using <a href="https://CRAN.R-project.org/package=data.table">data.table</a>.</li><li><a href="https://CRAN.R-project.org/package=SimInf">SimInf</a>: Provides an efficient and very flexible framework to conduct data-driven epidemiological modeling in realistic large scale disease spread simulations. The framework integrates infection dynamics in subpopulations as continuous-time Markov chains using the Gillespie stochastic simulation algorithm and incorporates available data such as births, deaths and movements as scheduled events at predefined time-points. Using C code for the numerical solvers and ‘OpenMP’ (if available) to divide work over multiple processors ensures high performance when simulating a sample outcome. One of our design goals was to make the package extendable and enable usage of the numerical solvers from other R extension packages in order to facilitate complex epidemiological research. The package contains template models and can be extended with user-defined models. For more details see the paper by <a href="https://doi.org/10.18637%2Fjss.v091.i12">Widgren, Bauer, Eriksson and Engblom (2019)</a>.</li><li><a href="https://CRAN.R-project.org/package=socialmixr">socialmixr</a>: Provides methods for sampling contact matrices from diary data for use in infectious disease modelling, as discussed in <a href="https://doi.org/10.1371%2Fjournal.pmed.0050074">Mossong et al. (2008)</a>.</li></ul><h3 id="transmission-tree-reconstruction">Transmission tree reconstruction</h3><ul><li><a href="https://CRAN.R-project.org/package=adegenet">adegenet</a>: while primarily a population genetics package, <a href="https://CRAN.R-project.org/package=adegenet">adegenet</a> also implements seqtrack (<a href="https://doi.org/10.1038%2Fhdy.2010.78">Jombart et al. 2011</a>), a maximum-parsimony approach for reconstructing transmission trees using the Edmonds/Chu-Liu algorithm</li><li><a href="https://CRAN.R-project.org/package=o2geosocial">o2geosocial</a> (<a href="https://CRAN.R-project.org/package=outbreaker2">outbreaker2</a> module): Bayesian reconstruction of who infected whom during past outbreaks using routinely-collected surveillance data. Inference of transmission trees using genotype, age specific social contacts, distance between cases and onset dates of the reported cases (<a href="https://doi.org/10.1098%2Frsif.2020.0084">Robert A, Kucharski AJ, Gastanaduy PA, Paul P, Funk S. 2020</a>).</li><li><a href="https://github.com/xavierdidelot/o2mod.transphylo">o2mod.transphylo</a> is a module of <a href="https://CRAN.R-project.org/package=outbreaker2">outbreaker2</a> which uses the <a href="https://CRAN.R-project.org/package=TransPhylo">TransPhylo</a> model of within-host evolution.</li><li><a href="https://CRAN.R-project.org/package=outbreaker2">outbreaker2</a>: Bayesian reconstruction of disease outbreaks using epidemiological and genetic information. <a href="https://doi.org/10.1371/journal.pcbi.1003457">Jombart T, Cori A, Didelot X, Cauchemez S, Fraser C and Ferguson N. 2014</a>, <a href="https://doi.org/10.1371/journal.pcbi.1006930">Campbell F, Cori A, Ferguson N, Jombart T. 2019</a>.</li><li><a href="https://CRAN.R-project.org/package=TransPhylo">TransPhylo</a>: Inference of transmission tree from a dated phylogeny. Includes methods to simulate and analyse outbreaks. The methodology is described in <a href="https://doi.org/10.1093%2Fmolbev%2Fmsu121">Didelot et al. (2014)</a>, <a href="https://doi.org/10.1093%2Fmolbev%2Fmsw275">Didelot et al. (2017)</a>.</li></ul><h3 id="vaccination">Vaccination</h3><ul><li><a href="https://CRAN.R-project.org/package=SCCS">SCCS</a>: Self-controlled case series models used to investigate associations between time-varying exposures such as vaccines or other drugs or non drug exposures and an adverse event can be fitted. Detailed information on the self-controlled case series method and its extensions with more examples at <a href="https://sccs-studies.info" class="uri">https://sccs-studies.info</a>.</li></ul><h2 id="helper-functions">Helper functions</h2><p>Here are packages providing useful helper functions for practicing, teaching and learning epidemiology (eg computing sample size, contingency tables, etc).</p><ul><li><a href="https://CRAN.R-project.org/package=DSAIDE">DSAIDE</a>: Exploration of simulation models (apps) of various infectious disease transmission dynamics scenarios. The purpose of the package is to help individuals learn about infectious disease epidemiology (ecology/evolution) from a dynamical systems perspective. All apps include explanations of the underlying models and instructions on what to do with the models.</li><li><a href="https://CRAN.R-project.org/package=epibasix">epibasix</a>: Contains elementary tools for analysis of common epidemiological problems, ranging from sample size estimation, through 2x2 contingency table analysis and basic measures of agreement (kappa, sensitivity/specificity). Appropriate print and summary statements are also written to facilitate interpretation wherever possible. The target audience includes advanced undergraduate and graduate students in epidemiology or biostatistics courses, and clinical researchers.</li><li><a href="https://CRAN.R-project.org/package=epiR">epiR</a>: Tools for the analysis of epidemiological and surveillance data. Contains functions for directly and indirectly adjusting measures of disease frequency, quantifying measures of association on the basis of single or multiple strata of count data presented in a contingency table, computation of confidence intervals around incidence risk and incidence rate estimates and sample size calculations for cross-sectional, case-control and cohort studies. Surveillance tools include functions to calculate an appropriate sample size for 1- and 2-stage representative freedom surveys, functions to estimate surveillance system sensitivity and functions to support scenario tree modelling analyses.</li><li><a href="https://CRAN.R-project.org/package=epitools">epitools</a>: Tools for training and practicing epidemiologists including methods for two-way and multi-way contingency tables.</li><li><a href="https://CRAN.R-project.org/package=epitrix">epitrix</a>: A collection of small functions useful for epidemics analysis and infectious disease modelling. This includes computation of basic reproduction numbers from growth rates, generation of hashed labels to anonymise data, and fitting discretised Gamma distributions.</li><li><a href="https://CRAN.R-project.org/package=powerSurvEpi">powerSurvEpi</a>: Functions to calculate power and sample size for testing main effect or interaction effect in the survival analysis of epidemiological studies (non-randomized studies), taking into account the correlation between the covariate of the interest and other covariates. Some calculations also take into account the competing risks and stratified analysis. This package also includes a set of functions to calculate power and sample size for testing main effects in the survival analysis of randomized clinical trials.</li></ul><h2 id="data">Data</h2><p>Here are packages providing different epidemiologic datasets, either simulated or real, useful for research purposes or field applications with a specific COVID-19 section.</p><h3 id="epidemic-outbreak-data">Epidemic outbreak data</h3><ul><li><a href="https://CRAN.R-project.org/package=contactdata">contactdata</a>: Data package for the supplementary data in <a href="https://doi.org/10.1371%2Fjournal.pcbi.1005697">Prem et al. (2017)</a>. Provides easy access to contact data for 152 countries, for use in epidemiological, demographic or social sciences research.</li><li><a href="https://CRAN.R-project.org/package=outbreaks">outbreaks</a>: Empirical or simulated disease outbreak data, provided either as RData or as text files.</li></ul><h3 id="covid-19">COVID-19</h3><ul><li><a href="https://CRAN.R-project.org/package=bets.covid19">bets.covid19</a>: Implements likelihood inference for early epidemic analysis. BETS is short for the four key epidemiological events being modeled: Begin of exposure, End of exposure, time of Transmission, and time of Symptom onset. The package contains a dataset of the trajectory of confirmed cases during the coronavirus disease (COVID-19) early outbreak. More detail of the statistical methods can be found in <a href="https://arxiv.org/abs/2004.07743">Zhao et al. (2020)</a>.</li><li><a href="https://CRAN.R-project.org/package=corona">corona</a>: Manipulate and view coronavirus data and other societally relevant data at a basic level.</li><li><a href="https://CRAN.R-project.org/package=coronavirus">coronavirus</a>: Provides a daily summary of the Coronavirus (COVID-19) cases by state/province. Data source: <a href="https://systems.jhu.edu/research/public-health/ncov/">Johns Hopkins University Center for Systems Science and Engineering (JHU CCSE) Coronavirus</a>.</li><li><a href="https://CRAN.R-project.org/package=COVID19">COVID19</a>: Download COVID-19 data across governmental sources at national, regional, and city level, as described in <a href="https://doi.org/10.21105/joss.02376">Guidotti and Ardia (2020)</a>. Includes the time series of vaccines, tests, cases, deaths, recovered, hospitalizations, intensive therapy, and policy measures by ‘<a href="https://www.bsg.ox.ac.uk/research/research-projects/coronavirus-government-response-tracker">Oxford COVID-19 Government Response Tracker</a>. Provides a seamless integration with’<a href="https://data.worldbank.org/">World Bank Open Data</a>‘,’<a href="https://www.google.com/covid19/mobility/">Google Mobility Reports</a>‘, ’Apple Mobility Reports](<a href="https://covid19.apple.com/mobility" class="uri">https://covid19.apple.com/mobility</a>)’.</li><li><a href="https://CRAN.R-project.org/package=covid19.analytics">covid19.analytics</a>: Load and analyze updated time series worldwide data of reported cases for the Novel CoronaVirus Disease (CoViD-19) from the <a href="https://github.com/CSSEGISandData/COVID-19">Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE) data repository</a>. The datasets are available in two main modalities, as a time series sequences and aggregated for the last day with greater spatial resolution. Several analysis, visualization and modelling functions are available in the package that will allow the user to compute and visualize total number of cases, total number of changes and growth rate globally or for an specific geographical location, while at the same time generating models using these trends; generate interactive visualizations and generate Susceptible-Infected-Recovered (SIR) model for the disease spread.</li><li><a href="https://CRAN.R-project.org/package=covid19br">covid19br</a>: Set of functions to import COVID-19 pandemic data into R. The Brazilian COVID-19 data, obtained from the official Brazilian repository at <a href="https://covid.saude.gov.br/" class="uri">https://covid.saude.gov.br/</a>, is available at country, region, state, and city-levels. The package also downloads the world-level COVID-19 data from the John Hopkins University’s repository.</li><li><a href="https://CRAN.R-project.org/package=covid19dbcand">covid19dbcand</a>: Provides different datasets parsed from ‘<a href="https://www.drugbank.ca/covid-19">Drugbank</a>’ database using <a href="https://CRAN.R-project.org/package=dbparser">dbparser</a> package. It is a smaller version from <a href="https://github.com/interstellar-Consultation-Services/dbdataset">dbdataset</a> package. It contains only information about COVID-19 possible treatment.</li><li><a href="https://CRAN.R-project.org/package=covid19italy">covid19italy</a>: Provides a daily summary of the Coronavirus (COVID-19) cases in Italy by country, region and province level. Data source: <a href="http://www.protezionecivile.it/">Presidenza del Consiglio dei Ministri - Dipartimento della Protezione Civile</a>.</li><li><a href="https://CRAN.R-project.org/package=covid19france">covid19france</a>: Imports and cleans <a href="https://github.com/opencovid19-fr/data" class="uri">https://github.com/opencovid19-fr/data</a> data on COVID-19 in France.</li><li><a href="https://github.com/nevrome/covid19germany">covid19germany</a>: An R package to load, visualise and analyse daily updated data on the COVID-19 outbreak in Germany.</li><li><a href="https://github.com/covid19R/covid19mobility">covid19mobility</a>: Scrapes trends in mobility after the Covid-19 outbreak from different sources. Currently, the package scrapes data from Google (<a href="https://www.google.com/covid19/mobility/" class="uri">https://www.google.com/covid19/mobility/</a>), Apple (<a href="https://www.apple.com/covid19/mobility" class="uri">https://www.apple.com/covid19/mobility</a>), and will add others. The data returned uses the tidy <a href="https://covid19r.github.io/documentation/">Covid19R project data standard</a> as well as the controlled vocabularies for measurement types.</li><li><a href="https://github.com/covid19R/covid19nytimes">covid19nytimes</a>: Accesses the NY Times Covid-19 county-level data for the US, described in <a href="https://www.nytimes.com/article/coronavirus-county-data-us.html" class="uri">https://www.nytimes.com/article/coronavirus-county-data-us.html</a> and available at <a href="https://github.com/nytimes/covid-19-data" class="uri">https://github.com/nytimes/covid-19-data</a>. It then returns the data in a tidy data format according to the Covid19R Project data specification. If you plan to use the data or publicly display the data or results, please make sure cite the original NY Times source. Please read and follow the terms laid out in the data license at <a href="https://github.com/nytimes/covid-19-data/blob/master/LICENSE" class="uri">https://github.com/nytimes/covid-19-data/blob/master/LICENSE</a>.</li><li><a href="https://CRAN.R-project.org/package=covid19sf">covid19sf</a>: Provides a verity of summary tables of the Covid19 cases in San Francisco. Data source: <a href="https://datasf.org/opendata/">San Francisco, Department of Public Health - Population Health Division</a>.</li><li><a href="https://CRAN.R-project.org/package=covid19swiss">covid19swiss</a>: Provides a daily summary of the Coronavirus (COVID-19) cases in Switzerland cantons and Principality of Liechtenstein. Data source: Specialist Unit for Open Government Data Canton of Zurich <a href="https://www.zh.ch/de/politik-staat/opendata.html" class="uri">https://www.zh.ch/de/politik-staat/opendata.html</a>.</li><li><a href="https://CRAN.R-project.org/package=covid19us">covid19us</a>: A wrapper around the ‘COVID Tracking Project API’ <a href="https://covidtracking.com/api/" class="uri">https://covidtracking.com/api/</a> providing data on cases of COVID-19 in the US.</li><li><a href="https://CRAN.R-project.org/package=CovidMutations">CovidMutations</a>: A feasible framework for mutation analysis and reverse transcription polymerase chain reaction (RT-PCR) assay evaluation of COVID-19, including mutation profile visualization, statistics and mutation ratio of each assay. The mutation ratio is conducive to evaluating the coverage of RT-PCR assays in large-sized samples. <a href="https://doi.org/10.20944/preprints202004.0529.v1">Mercatelli, D. and Giorgi, F. M. (2020)</a>.</li><li><a href="https://CRAN.R-project.org/package=oxcgrt">oxcgrt</a>: The Oxford COVID-19 Government Response Tracker (OxCGRT) tracks and compares worldwide government responses to the COVID-19 pandemic rigorously and consistently. OxCGRT makes available systematic information in a consistent way, aiding those who require information have access to it efficiently for their purposes. This package facilitates access to the OxCGRT data via its API <a href="https://covidtracker.bsg.ox.ac.uk/" class="uri">https://covidtracker.bsg.ox.ac.uk/</a> and includes functions to calculate the various OxCGRT indices in R.</li><li><a href="https://github.com/como-ph/oxcovid19">oxcovid19</a>: The <a href="https://covid19.eng.ox.ac.uk">OxCOVID19 Project</a> aims to increase our understanding of the COVID-19 pandemic and elaborate possible strategies to reduce the impact on the society through the combined power of statistical, mathematical modelling, and machine learning techniques. The OxCOVID19 Database is a large, single-centre, multimodal relational database consisting of information (using acknowledged sources) related to COVID-19 pandemic. This package provides an R-specific interface to the OxCOVID19 Database based on widely-used data handling and manipulation approaches in R.</li></ul><h2 id="links">Links</h2><ul><li><p>R Epidemics Consortium (RECON), non-profit organisation developing professional tools for field epidemiologists: <a href="https://www.repidemicsconsortium.org/projects/" class="uri">https://www.repidemicsconsortium.org/projects/</a></p></li><li><p>RECON’s task manager, issues regarding infectious epidemiology packages from RECON packages and partners: <a href="https://tasks.repidemicsconsortium.org" class="uri">https://tasks.repidemicsconsortium.org</a></p></li><li><p>Survival analysis task view: <a href="https://CRAN.R-project.org/view=Survival">Survival</a></p></li><li><p>Data visualisation and graphics task view: <a href="https://CRAN.R-project.org/view=Graphics">Graphics</a></p></li><li><p>Meta analysis task view: <a href="https://CRAN.R-project.org/view=MetaAnalysis">MetaAnalysis</a></p></li><li><p>Multivariate statistics task view: <a href="https://CRAN.R-project.org/view=Multivariate">Multivariate</a></p></li></ul>

  <h3>CRAN packages</h3>
  <table>
    <tr valign="top"><td><i>Core:</i></td><td><i>None</i>.</td></tr>
    <tr valign="top"><td><i>Regular:</i></td><td><a href="https://CRAN.R-project.org/package=adegenet">adegenet</a>, <a href="https://CRAN.R-project.org/package=argo">argo</a>, <a href="https://CRAN.R-project.org/package=bets.covid19">bets.covid19</a>, <a href="https://CRAN.R-project.org/package=cmprsk">cmprsk</a>, <a href="https://CRAN.R-project.org/package=contactdata">contactdata</a>, <a href="https://CRAN.R-project.org/package=corona">corona</a>, <a href="https://CRAN.R-project.org/package=coronavirus">coronavirus</a>, <a href="https://CRAN.R-project.org/package=COVID19">COVID19</a>, <a href="https://CRAN.R-project.org/package=covid19.analytics">covid19.analytics</a>, <a href="https://CRAN.R-project.org/package=covid19br">covid19br</a>, <a href="https://CRAN.R-project.org/package=covid19dbcand">covid19dbcand</a>, <a href="https://CRAN.R-project.org/package=covid19france">covid19france</a>, <a href="https://CRAN.R-project.org/package=covid19italy">covid19italy</a>, <a href="https://CRAN.R-project.org/package=covid19sf">covid19sf</a>, <a href="https://CRAN.R-project.org/package=covid19swiss">covid19swiss</a>, <a href="https://CRAN.R-project.org/package=covid19us">covid19us</a>, <a href="https://CRAN.R-project.org/package=CovidMutations">CovidMutations</a>, <a href="https://CRAN.R-project.org/package=data.table">data.table</a>, <a href="https://CRAN.R-project.org/package=dbparser">dbparser</a>, <a href="https://CRAN.R-project.org/package=dde">dde</a>, <a href="https://CRAN.R-project.org/package=deSolve">deSolve</a>, <a href="https://CRAN.R-project.org/package=dplyr">dplyr</a>, <a href="https://CRAN.R-project.org/package=DSAIDE">DSAIDE</a>, <a href="https://CRAN.R-project.org/package=earlyR">earlyR</a>, <a href="https://CRAN.R-project.org/package=endtoend">endtoend</a>, <a href="https://CRAN.R-project.org/package=Epi">Epi</a>, <a href="https://CRAN.R-project.org/package=epibasix">epibasix</a>, <a href="https://CRAN.R-project.org/package=epicontacts">epicontacts</a>, <a href="https://CRAN.R-project.org/package=EpiContactTrace">EpiContactTrace</a>, <a href="https://CRAN.R-project.org/package=EpiCurve">EpiCurve</a>, <a href="https://CRAN.R-project.org/package=epiDisplay">epiDisplay</a>, <a href="https://CRAN.R-project.org/package=EpiEstim">EpiEstim</a>, <a href="https://CRAN.R-project.org/package=epiflows">epiflows</a>, <a href="https://CRAN.R-project.org/package=EpiILM">EpiILM</a>, <a href="https://CRAN.R-project.org/package=EpiILMCT">EpiILMCT</a>, <a href="https://CRAN.R-project.org/package=epimdr">epimdr</a>, <a href="https://CRAN.R-project.org/package=EpiModel">EpiModel</a>, <a href="https://CRAN.R-project.org/package=epinet">epinet</a>, <a href="https://CRAN.R-project.org/package=epiR">epiR</a>, <a href="https://CRAN.R-project.org/package=EpiReport">EpiReport</a>, <a href="https://CRAN.R-project.org/package=episensr">episensr</a>, <a href="https://CRAN.R-project.org/package=epitools">epitools</a>, <a href="https://CRAN.R-project.org/package=epitrix">epitrix</a>, <a href="https://CRAN.R-project.org/package=etm">etm</a>, <a href="https://CRAN.R-project.org/package=i2extras">i2extras</a>, <a href="https://CRAN.R-project.org/package=incidence">incidence</a>, <a href="https://CRAN.R-project.org/package=incidence2">incidence2</a>, <a href="https://CRAN.R-project.org/package=mem">mem</a>, <a href="https://CRAN.R-project.org/package=memapp">memapp</a>, <a href="https://CRAN.R-project.org/package=mstate">mstate</a>, <a href="https://CRAN.R-project.org/package=nbTransmission">nbTransmission</a>, <a href="https://CRAN.R-project.org/package=nosoi">nosoi</a>, <a href="https://CRAN.R-project.org/package=o2geosocial">o2geosocial</a>, <a href="https://CRAN.R-project.org/package=odin">odin</a>, <a href="https://CRAN.R-project.org/package=outbreaker2">outbreaker2</a>, <a href="https://CRAN.R-project.org/package=outbreaks">outbreaks</a>, <a href="https://CRAN.R-project.org/package=oxcgrt">oxcgrt</a>, <a href="https://CRAN.R-project.org/package=pomp">pomp</a>, <a href="https://CRAN.R-project.org/package=popEpi">popEpi</a>, <a href="https://CRAN.R-project.org/package=powerSurvEpi">powerSurvEpi</a>, <a href="https://CRAN.R-project.org/package=R0">R0</a>, <a href="https://CRAN.R-project.org/package=riskCommunicator">riskCommunicator</a>, <a href="https://CRAN.R-project.org/package=RSurveillance">RSurveillance</a>, <a href="https://CRAN.R-project.org/package=SCCS">SCCS</a>, <a href="https://CRAN.R-project.org/package=SimInf">SimInf</a>, <a href="https://CRAN.R-project.org/package=socialmixr">socialmixr</a>, <a href="https://CRAN.R-project.org/package=SpatialEpi">SpatialEpi</a>, <a href="https://CRAN.R-project.org/package=surveillance">surveillance</a>, <a href="https://CRAN.R-project.org/package=TransPhylo">TransPhylo</a>, <a href="https://CRAN.R-project.org/package=trendeval">trendeval</a>, <a href="https://CRAN.R-project.org/package=trending">trending</a>, <a href="https://CRAN.R-project.org/package=tsiR">tsiR</a>.</td></tr>
  </table>


  <h3>Other resources</h3>
  <ul>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Graphics">Graphics</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=MetaAnalysis">MetaAnalysis</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Multivariate">Multivariate</a></li>
    <li>CRAN Task View: <a href="https://CRAN.R-project.org/view=Survival">Survival</a></li>
    <li>GitHub Project: <a href="https://github.com/nevrome/covid19germany">covid19germany</a></li>
    <li>GitHub Project: <a href="https://github.com/covid19R/covid19mobility">covid19mobility</a></li>
    <li>GitHub Project: <a href="https://github.com/covid19R/covid19nytimes">covid19nytimes</a></li>
    <li>GitHub Project: <a href="https://github.com/interstellar-Consultation-Services/dbdataset">dbdataset</a></li>
    <li>GitHub Project: <a href="https://github.com/epiforecasts/EpiNow2">EpiNow2</a></li>
    <li>GitHub Project: <a href="https://github.com/xavierdidelot/o2mod.transphylo">o2mod.transphylo</a></li>
    <li>GitHub Project: <a href="https://github.com/como-ph/oxcovid19">oxcovid19</a></li>
    <li>GitHub Project: <a href="https://github.com/reconhub/trendbreaker">trendbreaker</a></li>
  </ul>

</body>
</html>
